content/client-lib-development-guide/features.textile

```diff
@@ -1464,6 +1486,110 @@ enum StatsIntervalGranularity:
   DAY
   MONTH
 
+class DeviceDetails:
+  + getDeviceInfo() => io DeviceDetails
+  id: String
+  platform: DeviceType
+  type: DeviceType
+  clientId: String?
+  metadata: Dict<String, String>
+  updateTokenId: String
+  resetId(String)
+  resetUpdateToken() => io
+  push: DevicePushDetails
+
+class DevicePushDetails:
+  transportType: DevicePushTransportType
+  transportId: String
+  transportToken: String?
+  state: .Active | .Failing | .Failed
+  errorReason: ErrorInfo?
+  metadata: Dict<String, String>
+
+class Push:
+  publish(clientId: String, JsonObject) => io
+  publish(deviceId: String, JsonObject) => io
+  publish(transportType: String, transportId: String, transportTokenOrKey?: String) => io
+  admin: PushAdmin
+
+  // Only on platforms that support receiving notifications:
+
+  activate(DeviceDetails) => io ErrorInfo? // RSH2c
+  activateExternally(
```

Comment:
in place of `activeExternally`.

Ie `activate()` is called on the device.

`activateDevice()` is called by a server, because it wants to activate a device.